### PR TITLE
Merged FOBS from 2.1 to dev

### PR DESCRIPTION
**Pull request by nvidianz**

### CODE REVIEW

File: nvflare/private/fed/server/server_commands.py [hunk 1]
```diff
@@ -164,13 +164,14 @@ def process(self, data: Shareable, fl_ctx: FLContext):
...
+        shared_fl_ctx = data.get(ReservedKey.SHARED_FL_CONTEXT)
+        client = shareable.get_header(ServerCommandKey.FL_CLIENT)
         fl_ctx.set_peer_context(shared_fl_ctx)
-        contribution_task_name = data.get_header(ServerCommandKey.TASK_NAME)
-        task_id = data.get_cookie(FLContextKey.TASK_ID)
+        contribution_task_name = shareable.get_header(ServerCommandKey.TASK_NAME)
+        task_id = shareable.get_cookie(FLContextKey.TASK_ID)
         server_runner = fl_ctx.get_prop(FLContextKey.RUNNER)
-        server_runner.process_submission(client, contribution_task_name, task_id, data, fl_ctx)
+        server_runner.process_submission(client, contribution_task_name, task_id, shareable, fl_ctx)
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/server/server_commands.py:174]: why are we changing this section?

File: nvflare/fuel/utils/fobs/README.rst [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    fobs.register(SimpleDecomposer)
+    data = fobs.dumps(Simple(1, 'foo', datetime.now()))
+    obj = fobs.loads(data)
+    assert obj.num == 1
+    assert obj.name == 'foo'
+    assert isinstance(obj.timestamp, datetime)
+
+
+The same decomposer can be registered multiple times. Only first one takes effect, the others
+are ignored with a warning message.
```
Comment (YuanTingHsieh) [inline nvflare/fuel/utils/fobs/README.rst:148]: do we need to add a section about how to register decomposers in their custom codes?

Comment (nvidianz) [reply nvflare/fuel/utils/fobs/README.rst:148]: Good idea!

File: nvflare/private/fed/server/server_commands.py (hunk 1 above)
Comment (YuanTingHsieh) [reply nvflare/private/fed/server/server_commands.py:174]: Got it, if this has been tested then we are good.

**Review by YuanTingHsieh — APPROVED** [review]
thanks, LGTM

**Pull request merged**